### PR TITLE
feat(offline-datalog): allows user to config offset started logging time

### PR DESCRIPTION
**Pull request by Yuan3852**

### CODE REVIEW

File: src/views/OfflineDatalog.vue [hunk 1]
```diff
@@ -1,21 +1,39 @@
...
+      </div>
+    </ul>
+    <div class="datapicker">
+      <div class="datepciker-container">
+        <date-picker
+          v-model="dateTimeOffset"
+          type="datetime"
+          placeholder="Select datetime "
+          value-type="timestamp"
+          v-on:updateOption="timestamp"
```
Comment (MomePP) [inline src/views/OfflineDatalog.vue:22]: does this works ? may remove before merged

File: src/views/OfflineDatalog.vue [hunk 2]
```diff
@@ -1,21 +1,39 @@
...
-      {{ offlineDatalogStatus }}
-      {{ computePacket }}
+    <ul class="bt-container">
+      <div class="sync-container">
+        <button class="sync-bt" @click="syncOfflineDatalogRecords()">
+          Sync Data
+        </button>
+      </div>
+      <div class="delete-container">
+        <button class="delete-bt" @click="$vm2.open('modal')">
```
Comment (MomePP) [inline src/views/OfflineDatalog.vue:10]: when calling open/close modal from template, it seems no need to ref `$vm2` just like on [this example](https://github.com/initbase/vue-modal-2#optionscomponentname)

File: src/views/OfflineDatalog.vue [hunk 3]
```diff
@@ -91,15 +138,26 @@ export default {
...
+      //? Apply user datetime offset to series data.
+      let renderdata = structuredClone(
+        this.datalogRecords[this.selectedChannel["name"]]
+      ); //Deep Copy
+      renderdata.forEach((field) => {
+        for (let i = 0; i < field["data"].length; i++) {
+          field["data"][i][0] += this.dateTimeOffset;
+        }
+        return field["data"];
+      });
```
Comment (MomePP) [inline src/views/OfflineDatalog.vue:150]: is this the same method with `onSelectedDate` ?

File: src/views/OfflineDatalog.vue [hunk 4]
```diff
@@ -298,8 +356,31 @@ export default {
...
+      this.$refs.datalogChart.chartOptions.series = renderdata;
+    },
+    doDelete() {
+      this.$vm2.close("modal-1");
+      this.$vm2.close("modal-3");
+    },
+    close() {
+      this.$vm2.close("modal-4");
+    },
```
Comment (MomePP) [inline src/views/OfflineDatalog.vue]: not sure what these modals name mean ? can you describe how this being used ?

File: src/views/OfflineDatalog.vue (hunk 3 above)
Comment (MomePP) [inline src/views/OfflineDatalog.vue]: is this the same method with `onSelectedDate` ?

File: src/views/OfflineDatalog.vue (hunk 1 above)
Comment (MomePP) [inline src/views/OfflineDatalog.vue]: does `v-on:updateOption` works ? may remove before merged

File: src/views/OfflineDatalog.vue (hunk 2 above)
Comment (MomePP) [inline src/views/OfflineDatalog.vue:10]: when calling open/close modal from template, it seems no need to ref `$vm2` just like on [this example](https://github.com/initbase/vue-modal-2#optionscomponentname)

**Review by MomePP — CHANGES_REQUESTED** [review]
there are some questions, need to be refactors before merged

File: src/views/OfflineDatalog.vue (hunk 4 above)
Comment (Yuan3852) [reply src/views/OfflineDatalog.vue]: Sorry, It's both test functions. It can remove both

File: src/views/OfflineDatalog.vue (hunk 3 above)
Comment (Yuan3852) [reply src/views/OfflineDatalog.vue]: Yep it's same method with onSelectedDate

File: src/views/OfflineDatalog.vue (hunk 1 above)
Comment (Yuan3852) [reply src/views/OfflineDatalog.vue]: Yes , you can remove `v-on:updateOption`. it not work in datepicker

File: src/views/OfflineDatalog.vue (hunk 2 above)
Comment (Yuan3852) [reply src/views/OfflineDatalog.vue:10]: It need `$vm2` beacause in that example it has function close( ) in method , which in close ( ) function has `$vm2` for close modal.

**Comment by Yuan3852** [comment]
@MomePP I already resolve your comments in latest commit.

**Review by MomePP — APPROVED** [review]
Seem to be ok  👍

**Pull request merged**